chia/wallet/puzzles/did_innerpuz.clvm

```diff
@@ -9,16 +9,15 @@
   RECOVERY_DID_LIST_HASH  ; the list of DIDs that can send messages to you for recovery we store only the hash so that we don't have to reveal every time we make a message spend
   NUM_VERIFICATIONS_REQUIRED  ; how many of the above list are required for a recovery
   SINGLETON_STRUCT  ; my singleton_struct, formerly a Truth - ((SINGLETON_MOD_HASH, (LAUNCHER_ID, LAUNCHER_PUZZLE_HASH)))
-  DID_MOD_HASH  ; DID inner puzzle treehash
   METADATA ; Customized metadata, e.g KYC info
-  mode  ; this indicates which spend mode we want. 0. Recovery mode 1. Send message mode 2. Transfer mode
-  new_amount  ; DIDs can receive payments so when we recreate ourselves sometimes we want to change our amount
-  p2_solution ; Solution of the inner P2 puzzle, only required in the create message mode and transfer mode.
-  new_inner_puzhash  ; In message mode, this should be the DID puzzle hash. In the transfer case, this should be the recipient p2 puzzle hash. In recovery mode, this will be the new wallet DID puzzle hash
+  mode  ; this indicates which spend mode we want. 0. Recovery mode 1. Run P2_PUZZLE with p2_solution
+  new_amount_or_p2_solution  ; In mode 0, maybe we want to change our amount
```

Comment:
How is new_amount secured in this scenario?  It seems like a malicious node would always change it to zero.

Reply:
Ok yeah, let's just lock this down to our actual amount and do an assert_my_amount